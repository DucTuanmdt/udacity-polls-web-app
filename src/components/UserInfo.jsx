import React from "react";

function UserInfo({ name, avatar }) {
  return (
    <div className="d-flex align-items-center gap-2">
      <img
        src={avatar}
        alt="avatar"
        width={40}
        height={40}
        className="rounded-pill"
      />
      <span>{name}</span>
    </div>
  );
}

export default UserInfo;
