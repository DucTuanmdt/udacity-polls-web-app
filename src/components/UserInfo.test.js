import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import UserInfo from "./UserInfo";

const defaultProps = {
  avatar: "https://i.pravatar.cc/150?img=1",
  name: "Peter",
};
describe("Snapshot test for UserInfo.jsx", () => {
  test("should render correctly", () => {
    const tree = renderer.create(<UserInfo {...defaultProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("should render avatar and user's name", () => {
    render(<UserInfo {...defaultProps} />);
    expect(screen.getByRole("img")).toBeInTheDocument();
    expect(screen.getByText(defaultProps.name)).toBeInTheDocument();
  });
});
