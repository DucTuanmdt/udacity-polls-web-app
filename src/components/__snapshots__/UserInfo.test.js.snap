// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test for UserInfo.jsx should render correctly 1`] = `
<div
  className="d-flex align-items-center gap-2"
>
  <img
    alt="avatar"
    className="rounded-pill"
    height={40}
    src="https://i.pravatar.cc/150?img=1"
    width={40}
  />
  <span>
    Peter
  </span>
</div>
`;
